src/Microsoft.Azure.SignalR.Common/Auth/AuthUtility.cs

```diff
@@ -65,8 +65,12 @@ public static string GenerateAccessToken(byte[] keyBytes,
         return jwtToken.Length > MaxTokenLength ? throw new AzureSignalRAccessTokenTooLongException() : jwtToken;
     }
 
-    public static string GenerateRequestId()
+    public static string GenerateRequestId(string traceIdentifier)
     {
-        return Convert.ToBase64String(BitConverter.GetBytes(Stopwatch.GetTimestamp()));
+        // Before filled into query string, this id will be process by "WebUtility.UrlEncode(...)". So base64 encoding is not needed.
+        // Use hex to shorten the length.
+        return string.IsNullOrEmpty(traceIdentifier) 
+            ? Stopwatch.GetTimestamp().ToString("X")
```

Comment:
following the practice of traceIdentifier, we could also use some interlock.increment index appended to the timestamp to make it unique